src/functional/ffront/decorator.py

```diff
@@ -94,6 +94,32 @@ def _filter_closure_vars_by_type(closure_vars: dict[str, Any], *types: type) ->
     return {name: value for name, value in closure_vars.items() if isinstance(value, types)}
 
 
+def _canonicalize_args(
+    node_params: list[ct.FieldType | ct.ScalarType],
+    args: tuple[ct.SymbolType],
+    kwargs: dict[str, ct.SymbolType],
+) -> tuple[tuple, dict]:
+    new_args = []
+    new_kwargs = {**kwargs}
+
+    for param_i, param in enumerate(node_params):
+        if param.id in new_kwargs:
+            if param_i < len(args):
+                raise ProgramTypeError(f"got multiple values for argument {param.id}.")
+            new_args.append(kwargs[param.id])
+            new_kwargs.pop(param.id)
+        elif param_i < len(args):
+            new_args.append(args[param_i])
+        else:
+            # case when param in function definition but not in function call
+            # e.g. function expects 3 parameters, but only 2 were given.
+            # Error covered later in `accept_args`.
```

Comment:
Why did you remove the part of the sentence that motivates why errors are not covered here?

Reply:
I merely re-phrased, it still says: "Error covered later in `accept_args`."